src/App.tsx

```diff
@@ -9,38 +9,40 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as postsActions } from './features/postsSlice';
+import { actions as selectedPostActions } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { author } = useAppSelector(state => state.author);
+  const { posts, loading, error } = useAppSelector(state => state.posts);
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const dispatch = useAppDispatch();
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
+    dispatch(postsActions.setLoading(true));
 
     getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+      .then(postsFromServer => {
+        dispatch(postsActions.set(postsFromServer));
+      })
+      .catch(() => {
+        dispatch(postsActions.setError('Something went wrong'));
+      })
+      .finally(() => {
+        dispatch(postsActions.setLoading(false));
+      });
   }
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(selectedPostActions.set(null));
 
     if (author) {
       loadUserPosts(author.id);
     } else {
-      setPosts([]);
+      dispatch(postsActions.set([]));
     }
-  }, [author]);
+  }, [author, dispatch]);
```

Comment:
```suggestion
  }, [author, dispatch, loadUserPosts]);
```

Reply:
in working Demo  we can to add a comment with spaces only
I thought that I have to change task just with Redux or also i have to change logic of component - NewCommentForm?